src/sensai/featuregen.py

```diff
@@ -45,6 +45,7 @@ def __init__(self, categoricalFeatureNames: Optional[Union[Sequence[str], str]]
             raise ValueError(f"normalisationRules should be empty when a normalisationRuleTemplate is provided")
 
         self._generatedColumnNames = None
+        self.categoricalFeatureNames = categoricalFeatureNames
```

Comment:
sure, I just wanted it for the summary. 

Short story on why I did it: I don't always make things private assuming "the user will know not to change it". I remember that for some base classes in sklearn all parameters should be saved as public fields of the same name (forgot what for now), so the habit sticked

Reply:
It's not only about knowing not to change it but also about assuming what it contains. Even reading it could lead to problems. A user might think it's always a list of column names. Since you can't provide a docstring for an attribute (what a broken language!), users can't even safely read the property.

Reply:
What surprises me here is that you didn't find it aesthetically unpleasing or inconsistent to add a new property which is now the only property that isn't private :)
scnr